pkg/api/scylla/validation/cluster_validation.go

```diff
@@ -32,90 +33,30 @@ var (
 		AlternatorWriteIsolationOnlyRMWUsesLWT,
 	}
 
-	SupportedBroadcastAddressTypes = []scyllav1.BroadcastAddressType{
+	SupportedScyllaV1BroadcastAddressTypes = []scyllav1.BroadcastAddressType{
 		scyllav1.BroadcastAddressTypePodIP,
 		scyllav1.BroadcastAddressTypeServiceClusterIP,
 		scyllav1.BroadcastAddressTypeServiceLoadBalancerIngress,
 	}
 
+	SupportedScyllaV1Alpha1BroadcastAddressTypes = []scyllav1alpha1.BroadcastAddressType{
+		scyllav1alpha1.BroadcastAddressTypePodIP,
+		scyllav1alpha1.BroadcastAddressTypeServiceClusterIP,
+		scyllav1alpha1.BroadcastAddressTypeServiceLoadBalancerIngress,
+	}
+
 	schedulerTaskSpecCronParseOptions = cron.SecondOptional | cron.Minute | cron.Hour | cron.Dom | cron.Month | cron.Dow | cron.Descriptor
 )
 
 func ValidateScyllaCluster(c *scyllav1.ScyllaCluster) field.ErrorList {
-	return ValidateScyllaClusterSpec(&c.Spec, field.NewPath("spec"))
-}
-
-func ValidateUserManagedTLSCertificateOptions(opts *scyllav1.UserManagedTLSCertificateOptions, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if len(opts.SecretName) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("secretName"), ""))
-	} else {
-		for _, msg := range apimachineryvalidation.NameIsDNSSubdomain(opts.SecretName, false) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("secretName"), opts.SecretName, msg))
-		}
-	}
-
-	return allErrs
-}
-
-func ValidateOperatorManagedTLSCertificateOptions(opts *scyllav1.OperatorManagedTLSCertificateOptions, fldPath *field.Path) field.ErrorList {
-	allErrs := field.ErrorList{}
-
-	for _, dnsName := range opts.AdditionalDNSNames {
-		for _, msg := range apimachineryutilvalidation.IsDNS1123Subdomain(dnsName) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("additionalDNSNames"), opts.AdditionalDNSNames, msg))
-		}
-	}
-
-	for _, ip := range opts.AdditionalIPAddresses {
-		for _, msg := range apimachineryutilvalidation.IsValidIP(ip) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("additionalIPAddresses"), opts.AdditionalIPAddresses, msg))
-		}
+	_, err := scyllacluster.MigrateV1ScyllaClusterSpecToV1Alpha1ScyllaDBDatacenterSpec(c.Name, c.Spec)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(field.NewPath("spec"), fmt.Errorf("can't migrate v1.ScyllaCluster.Spec to v1alpha1.ScyllaDBDatacenter.Spec: %w", err)))
```

Comment:
this looks prone to tightening the validation which is strictly forbidden - does it need to retun the error?

Reply:
indeed, removed.